scene/2d/navigation_agent_2d.cpp

```diff
@@ -322,6 +331,10 @@ Vector2 NavigationAgent2D::get_next_location() {
 	}
 }
 
+Vector<Vector2> NavigationAgent2D::get_nav_path() const {
+	return navigation_result->get_path();
+}
+
```

Comment:
Against, consider whether this should be a const reference (I'm not offhand absolutely sure it can be bound as const reference, so that might be deciding factor).

Reply:
I'm not so sure about this change.  Yes, for C++ it makes sense since `const` will prevent edits.  However, C#, GDScript, and other languages may not pass the `const`ness through the binding, allowing users to change the path when they think they are getting a copy.

I think that behavior would be surprising, as I believe most users of this API would believe they are getting a copy of the path, not a reference to it.  Under that premise, I believe we should return a Vector and allow the COW functionality to kick in if changes are made.

Reply:
This should be tested imo. If binding loses const correctness, this would be a major bug / something in need of documentation.

It is also possible that binding a const ref is simply not allowed by the binding system, in which case this would be resolved.

Reply:
I did some testing and can confirm that GDScript does not modify the path when it is returned as a `const Vector<Vector2>&`.  Any calls to mutate the `Vector<T>` appear to do nothing (but no warning is printed).

It still feels a bit off to me, but since I'm not an expert in C++ I went ahead and took your suggestion and made the change.